src/background_thread.c

```diff
@@ -640,9 +647,12 @@ load_pthread_create_fptr(void) {
 	if (pthread_create_fptr) {
 		return pthread_create_fptr;
 	}
-
+#ifdef JEMALLOC_LAZY_LOCK
+	isthreaded = true;
+#endif
 	pthread_create_fptr = dlsym(RTLD_NEXT, "pthread_create");
-	if (pthread_create_fptr == NULL) {
+	real_pthread_create = dlsym(RTLD_DEFAULT, "pthread_create");
```

Comment:
This naming seems backwards as compared to the meaning used for "real" in the ```dlsym(3)``` manual page.  ```RTLD_DEFAULT``` finds the top symbol in the stack, i.e. the override, whereas ```RTLD_NEXT``` finds the "real" symbol that was overridden.  Given the naming mixup, I can't tell what the actual intention is here.